chain/mainnet/mezo_31612-1/genesis.json

```diff
@@ -1 +1,296 @@
+        },
+        {
+          "@type": "/ethermint.types.v1.EthAccount",
+          "base_account": {
+            "address": "mezo1qcjuy52698gf7xaymfdg00guh3nnse6chm0s2t",
+            "pub_key": null,
+            "account_number": "1",
+            "sequence": "0"
+          },
+          "code_hash": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+        },
+        {
+          "@type": "/ethermint.types.v1.EthAccount",
+          "base_account": {
+            "address": "mezo14ugypmyfn2whx4dhsw4jwflk2x5fx2vkvmjrzk",
+            "pub_key": null,
+            "account_number": "2",
+            "sequence": "0"
+          },
+          "code_hash": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+        },
+        {
+          "@type": "/ethermint.types.v1.EthAccount",
+          "base_account": {
+            "address": "mezo1lgrfdvcj38gwcwr0qv4ytdfcygc0mg6slsj3yt",
+            "pub_key": null,
+            "account_number": "3",
+            "sequence": "0"
+          },
+          "code_hash": "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"
+        }
+      ]
+    },
+    "authz": {
+      "authorization": []
+    },
+    "bank": {
+      "params": {
+        "send_enabled": [],
+        "default_send_enabled": true
+      },
+      "balances": [
+        {
+          "address": "mezo1qcjuy52698gf7xaymfdg00guh3nnse6chm0s2t",
+          "coins": [
+            {
+              "denom": "amezo",
+              "amount": "200000000000000000000000000"
+            }
+          ]
+        },
+        {
+          "address": "mezo188vx087k295fxdlx7f4wkznmghajnf0maysxpe",
+          "coins": [
+            {
+              "denom": "amezo",
+              "amount": "300000000000000000000000000"
+            }
+          ]
+        },
+        {
+          "address": "mezo14ugypmyfn2whx4dhsw4jwflk2x5fx2vkvmjrzk",
+          "coins": [
+            {
+              "denom": "amezo",
+              "amount": "400000000000000000000000000"
+            }
+          ]
+        },
+        {
+          "address": "mezo1lgrfdvcj38gwcwr0qv4ytdfcygc0mg6slsj3yt",
+          "coins": [
+            {
+              "denom": "amezo",
+              "amount": "100000000000000000000000000"
+            }
+          ]
+        }
+      ],
+      "supply": [
+        {
+          "denom": "amezo",
+          "amount": "1000000000000000000000000000"
+        }
+      ],
+      "denom_metadata": [],
+      "send_enabled": []
+    },
+    "bridge": {
+      "params": {
+        "max_erc20_tokens_mappings": 20,
+        "btc_supply_assertion_enabled": true
+      },
+      "assets_locked_sequence_tip": "0",
+      "source_btc_token": "0x18084fbA666a33d37592fA2633fD49a74DD93a88",
+      "erc20_tokens_mappings": [],
+      "initial_btc_supply": "0"
+    },
+    "crisis": {
+      "constant_fee": {
+        "denom": "abtc",
+        "amount": "1000"
+      }
+    },
+    "evm": {
```

Comment:
What do we do about pre-deployment contracts? To save us operational pain later, we could deploy ERC-2470 singleton factory and Multicall3.

We will also need to through the SafeFactory deployment logs and see if the transactions were pre-signed and require a bag of BTC to deploy.

Reply:
Thinking more about it, the problem is more complicated than it seems. We will have to deploy not only Safe contracts with all their dependencies, but also OrangeKit contracts via an upgrade to v1.0.0-beta.3 per https://github.com/mezo-org/passport/pull/86 to maintain stable addresses. It feels that the best solution is to just return the chain fees once we are done.

Reply:
Yes, pre-deployment of such contracts will be complicated and error prone as we need to work on bare byte code. The refund of chain fees through the splitter address is far easier.